darts/models/forecasting/regression_model.py

```diff
@@ -393,7 +393,9 @@ def _generate_lags(
                 processed_lags[lags_abbrev] = [
                     lag_ + output_chunk_shift for lag_ in processed_lags[lags_abbrev]
                 ]
-                if processed_component_lags:
+                if processed_component_lags and list(tmp_components_lags.keys()) != [
+                    "default_lags"
+                ]:
```

Comment:
You could store this under a variable to avoid recomputing the condition (replace it above L385 and below):

```
has_default_lags = list(tmp_components_lags.keys()) == ["default_lags"]
```


```suggestion
                if processed_component_lags and not has_default_lags:
```